src/vm/eventtrace.cpp

```diff
@@ -7347,6 +7403,140 @@ VOID ETW::EnumerationLog::EnumerationHelper(Module *moduleFilter, BaseDomain *do
     }
 }
 
+bool ETW::TieredCompilationLog::Runtime::IsEnabled()
```

Comment:
You should move the IsEnabled functions to the header file and mark them as FORCEINLINE.  Ideally the cost for these is super-low and in fact all of the code in these functions except for the contract will just turn into an inlined bit check.

Reply:
I had tried something like that initially, there were some dependency issues that sort of looked like it might need an .inl file to resolve, didn't look too closely. Will take another look and will add .inl if necessary.